### PR TITLE
API improvement

### PR DESCRIPTION
**Pull request by FrainL**
## Description

1. Rename Item to Model
2. Merge two Adapter into one
3. Simplify APIs
4. Use result builder for list-building

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/Alpensegler/.github//blob/master/CONTRIBUTING.md) guide
- [x] I added a very descriptive title to this pull request
- [x] I have follow the [guide](https://www.conventionalcommits.org/en/v1.0.0/) to write commit messages

### CODE REVIEW

File: Sources/ListAdapter/AnyListSourcesBuilder.swift [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    }
+}
+
+@resultBuilder
+public struct AnyListSourcesBuilder<View: ListView> {
+    public static func buildIf<S: ListAdapter>(_ content: S?) -> S? where S.View == View {
+        content
+    }
+
+    public static func buildEither<TrueContent, FalseContent>(
```
Comment (Cokile) [inline Sources/ListAdapter/AnyListSourcesBuilder.swift:29]: Should we use `buildEither<TrueContent: ListAdapter, FalseContent: ListAdapter>` to make where clause more succinct?

**Review by Cokile — COMMENTED** [review]
Very brilliant change! Looks f**king good to me!

File: Sources/Context/ListContext.swift [hunk 2]
```diff
@@ -10,22 +10,31 @@ import Foundation
 @dynamicMemberLookup
 public protocol Context {
     associatedtype Base: DataSource
-    associatedtype List
+    associatedtype View
 
     var source: Base.SourceBase.Source { get }
-    var listView: List { get }
+    var listView: View { get }
```
[the remark this answers is not in the document]
Comment (FrainL) [reply Sources/Context/ListContext.swift:16]: If we wanna use `ListView` here, in the future, if we want to make sure the associated type `ListView` conform to protocol `ListView`, which is probably what we will do next, we will need to use `ListKit.ListView`. I prefer we keep View here.

File: Sources/Coordinator/ListCoordinator.swift [hunk 3]
```diff
@@ -77,33 +77,34 @@ public class ListCoordinator<SourceBase: DataSource> where SourceBase.SourceBase
...
-        object: Object,
+        view: V,
         with input: Input,
+        index: Index,
         _ offset: Index
     ) -> Output? {
-        guard let rawClosure = context.listDelegate.functions[function.selector],
-              let closure = rawClosure as? (Object, Context, CoordinatorContext, Input, Index, Index) -> Output
+        guard let rawClosure = context.listDelegate.functions[selector],
+              let closure = rawClosure as? (AnyObject, Context, CoordinatorContext, Input, Index, Index) -> Output
```
[the remark this answers is not in the document]
Comment (FrainL) [reply Sources/Coordinator/ListCoordinator.swift:105]: Maybe not, the closure itself is `(AnyObject, Context, CoordinatorContext, Input, Index, Index) -> Output`, which is defined in `ListDelegate.Function`, does not contain the certain type of View.

File: Sources/Coordinator/ListCoordinator.swift [hunk 4]
```diff
@@ -77,33 +77,34 @@ public class ListCoordinator<SourceBase: DataSource> where SourceBase.SourceBase
     func configExtraSelector(delegate: ListDelegate) -> Set<Selector>? { nil }
 
     @discardableResult
-    func apply<Object: AnyObject, Target, Input, Output, Closure>(
-        _ function: ListDelegate.Function<Object, Delegate, Target, Input, Output, Closure>,
+    func apply<V: AnyObject, Input, Output>(
```
[the remark this answers is not in the document]
Comment (FrainL) [reply Sources/Coordinator/ListCoordinator.swift]: Changed to AnyObject instead of generic type.

**Pull request merged**